lib/fleiss/worker.rb

```diff
@@ -62,34 +60,37 @@ def run_cycle
 
     batch.each do |job|
       @pool.post do
-        Fleiss.backend.wrap_perform { perform(job) }
+        perform(job)
       end
     end
   rescue StandardError => e
-    handle_exception e, 'running cycle'
+    log_exception e, 'running cycle'
   end
 
   def perform(job)
     thread_id = Thread.current.object_id.to_s(16).reverse
     owner     = "#{uuid}/#{thread_id}"
-    return unless job.start(owner)
 
-    log(:info) { "Worker #{uuid} execute job ##{job.id} (by thread #{thread_id})" }
-    finished = false
-    begin
-      ActiveJob::Base.execute job.job_data
-      finished = true
-    rescue StandardError
-      finished = true
-      raise
-    ensure
-      finished ? job.finish(owner) : job.reschedule(owner)
+    Fleiss.backend.wrap_perform do
+      return unless job.start(owner)
+
+      log(:info) { "Worker #{uuid} execute job ##{job.id} (by thread #{thread_id})" }
+      finished = false
+      begin
+        ActiveJob::Base.execute job.job_data
```

Comment:
Can potentially raise err on DB conn failure if job uses ActiveRecord.

Maybe we shouldn't capture these exceptions by wrap_perform, but IMO it's OK.